src/components/CodeEditor/PythonEditor.js

```diff
@@ -0,0 +1,154 @@
+import RestartAltIcon from "@mui/icons-material/RestartAlt";
+import CableIcon from "@mui/icons-material/Cable";
+
+import { Box, Button, Grid, Typography } from "@mui/material";
+
+const PythonEditor = ({
+  initialCodeEditorValue,
+  disableEditing,
+  disableRun,
+}) => {
+  let { runPython, stdout, stderr, isLoading, isRunning } = usePython();
+  const [pythonEditorCode, setPythonEditorCode] = useState(
+    initialCodeEditorValue
+  );
+  const [codeRan, setCodeRan] = useState();
+  let [codeExecuted, setCodeExecuted] = useState(stdout, stderr);
+
+  useEffect(() => {
+    if (stdout) {
+      setCodeExecuted(stdout);
+    } else {
+      setCodeExecuted(stderr);
+    }
+  }, [stdout, stderr]);
+  return (
+    <Box
+      className="PythonCodeEditor"
+      sx={{
+        my: 3,
+      }}
+    >
+      <Box
+        sx={{
+          p: "2px",
+          border: "1px solid #6D6D6D",
+          borderRadius: "8px",
+          mb: 1.5,
+        }}
+      >
+        <CodeMirrorEditor
+          value={pythonEditorCode}
+          setEditorState={setPythonEditorCode}
+          readOnly={disableEditing}
+        />
+        <Box
+          className="middle-border"
+          sx={{ margin: "0 -2px", borderTop: "1px solid #6D6D6D" }}
+        />
+
+        {disableRun ? (
+          <Typography sx={{ p: 1, color: "#6D6D6D" }}>
+            Code not runnable? It's because this code does not have output
+            statements and is meant for understanding the code snippet
+          </Typography>
+        ) : (
+          <Box>
+            {disableEditing && (
+              <Typography
+                sx={{
+                  p: 1,
+                  color: "#6D6D6D",
+                  margin: "0 -2px",
+                  borderBottom: "1px solid #6D6D6D",
+                }}
+              >
+                Code not editable? It's because we want you to understand the
+                outcome directly
+              </Typography>
+            )}
+            <Box
+              className="toolbar"
+              sx={{
+                display: "flex",
+                justifyContent: "space-between",
+                p: 2,
+              }}
+            >
+              <Button
+                startIcon={<RestartAltIcon />}
+                disabled={
+                  isLoading ||
+                  (initialCodeEditorValue === pythonEditorCode && !codeExecuted)
+                }
+                variant="outlined"
+                onClick={() => {
+                  console.log(initialCodeEditorValue);
+                  setPythonEditorCode(initialCodeEditorValue);
+                  setCodeExecuted("");
+                }}
+              >
+                Reset
+              </Button>
+              <Button
+                endIcon={<ArrowRightIcon />}
+                disabled={
+                  isLoading ||
+                  isRunning ||
+                  (codeRan === pythonEditorCode && codeExecuted)
+                }
```

Comment:
Why does `(codeRan === pythonEditorCode && codeExecuted)` have any determination in whether the button is disabled here? It seems to me that you'd only want to disable it when it's loading or the code is running.
```suggestion
                disabled={isLoading || isRunning}
```